src/main/commands/commandManager.ts

```diff
@@ -274,7 +275,7 @@ export class CommandManager implements ExtensionComponent {
     private async doConnect() {
         let credentialsSet: boolean = await this._connectionManager.connect();
         if (credentialsSet) {
-            this.doRefresh(true);
+            await this.doRefresh(true);
```

Comment:
We should verify that the "await" does not hang the UI. Same for the `doDisconnect`.